### PR TITLE
Bug 1544616 - better support for building base

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -237,14 +231,14 @@ are important.
       using it.. Usually you need to also update a provisioner workerType with
       the new AMI id.
 
-      To do this, run `./deploy.sh`, which gathers the necessary secrets and, on `https://taskcluster.net`, deploys to worker types.
-      The secrets come from files in passwordstore named after the deploymnet, which is determined from `TASKCLUSTER_ROOT_URL`.
+To do this, run `./deploy.sh <target>`, which gathers the necessary secrets and, on `https://taskcluster.net`, deploys to worker types.
+The secrets come from files in passwordstore named after the deploymnet, which is determined from `TASKCLUSTER_ROOT_URL`.
```
Comment (walac) [inline README.md]: nit: s/deploymnet/deployment/

File: deploy.sh [hunk 2]
```diff
@@ -69,7 +72,7 @@ fi
 
 deploy/bin/import-docker-worker-secrets
 trap 'rm -rf /tmp/docker-worker*' EXIT
-deploy/bin/build app
+deploy/bin/build $BUILD_TARGET
 if [ "$DEPLOYMENT" == "taskcluster-net" ]; then
     deploy/bin/update-worker-types.js $*
```
Comment (walac) [inline deploy.sh]: This only makes sense for `app`, doesn't it?

Comment (djmitche) [reply deploy.sh]: Good point!

**Comment by djmitche** [comment]
OK, merge when you're ready -- I've already gotten the benefit I need out of it :)

**Pull request merged**